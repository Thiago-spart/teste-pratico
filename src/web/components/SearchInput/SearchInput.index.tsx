import type { FC, InputHTMLAttributes } from "react";
import { BiSearch } from "react-icons/bi";

import * as S from "./SearchInput.styled";

export const SearchInput: FC<InputHTMLAttributes<HTMLInputElement>> = ({
	type = "search",
	...rest
}) => {
	return (
		<S.Container>
			<S.SearchInputStyle type={type} {...rest} />
			<BiSearch />
		</S.Container>
	);
};
