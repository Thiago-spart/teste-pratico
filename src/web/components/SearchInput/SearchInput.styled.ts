import styled from "styled-components";

export const Container = styled.label`
	display: flex;
	align-items: flex-start;
	justify-content: flex-start;
	position: relative;
	width: 100%;

	svg {
		position: absolute;
		font-size: 2.2rem;
		pointer-events: none;
		top: 12px;
		left: 10px;
	}
`;

export const SearchInputStyle = styled.input`
	font-weight: 300;
	font-size: 1.6rem;
	line-height: 2rem;
	letter-spacing: 0.15rem;
	border-radius: 0.75rem;
	padding: 1.2rem 1rem 1.2rem 4rem;
	transition: 0.2s;
	width: 100%;

	&:focus-visible {
		outline: none;
		box-shadow: rgba(0, 0, 0, 0.35) 0px 5px 15px;
	}

	&[type="search"]::-webkit-search-cancel-button {
		-webkit-appearance: none;
		height: 1.8rem;
		width: 1.8rem;
		border-radius: 50%;
		background: url(https://pro.fontawesome.com/releases/v5.10.0/svgs/solid/times-circle.svg)
			no-repeat 50% 50%;
		background-size: contain;
		opacity: 0;
		pointer-events: none;
	}

	&[type="search"]:focus::-webkit-search-cancel-button {
		opacity: 0.6;
		pointer-events: all;
		cursor: pointer;
	}
`;
