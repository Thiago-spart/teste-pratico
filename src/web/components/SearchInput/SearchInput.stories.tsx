/* eslint-disable import/no-default-export */
/* eslint-disable import/exports-last */
import type { ComponentStory, Meta } from "@storybook/react";

import { SearchInput } from "./SearchInput.index";

const meta: Meta<typeof SearchInput> = {
	title: "Componentes/Input/Search",
	component: SearchInput,
};

const Template: ComponentStory<typeof SearchInput> = props => (
	<SearchInput {...props} />
);

export const Default = Template.bind({});
Default.args = {
	placeholder: "Search...",
};

export default meta;
