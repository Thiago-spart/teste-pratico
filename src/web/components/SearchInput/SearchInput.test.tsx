import { composeStories } from "@storybook/testing-react";
import { render, screen } from "@testing-library/react";

import userEvent from "@testing-library/user-event";

import * as stories from "./SearchInput.stories";

const { Default } = composeStories(stories);

describe("Button component", () => {
	it("should match snapshot", () => {
		const { container } = render(<Default />);

		expect(container).toMatchSnapshot();
	});

	it("should be filled up", () => {
		render(<Default />);

		const inputElement = screen.getByPlaceholderText("Search...");

		userEvent.type(inputElement, "London");

		expect(screen.findByText("London")).toBeTruthy();
	});
});
