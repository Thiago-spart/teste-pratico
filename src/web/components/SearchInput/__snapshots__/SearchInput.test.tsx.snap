// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should match snapshot 1`] = `
<div>
  <label
    class="sc-bczRLJ gjYiOx"
  >
    <input
      class="sc-gsnTZi dAHsDJ"
      placeholder="Search..."
      type="search"
    />
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10 18a7.952 7.952 0 0 0 4.897-1.688l4.396 4.396 1.414-1.414-4.396-4.396A7.952 7.952 0 0 0 18 10c0-4.411-3.589-8-8-8s-8 3.589-8 8 3.589 8 8 8zm0-14c3.309 0 6 2.691 6 6s-2.691 6-6 6-6-2.691-6-6 2.691-6 6-6z"
      />
    </svg>
  </label>
</div>
`;
